wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -515,42 +514,29 @@ public final class MLSService: MLSServiceInterface {
         }
     }
 
-    private func claimKeyPackages(for users: [MLSUser]) async throws -> [KeyPackage] {
-        logger.info("claiming key packages for users: \(users)")
-        do {
-            guard let context = context else { return [] }
+    private func claimKeyPackages(
+        for users: [MLSUser]
+    ) async -> [KeyPackage] {
+
+        guard let context = context else { return [] }
 
-            var result = [KeyPackage]()
+        var result = [KeyPackage]()
 
-            for try await keyPackages in claimKeyPackages(for: users, in: context) {
+        for user in users {
+            do {
+                let keyPackages = try await actionsProvider.claimKeyPackages(
```

Comment:
I noticed that we don't throw an error if a user a has run out out of key packages, the backend will then return an empty array of key packages. I think we should so that we can correctly inform why a user couldn't be added.

Reply:
Empty key package list is now handled as an error